File: 2348. Number of Zero-Filled Subarrays.kt
class Solution {
    fun zeroFilledSubarray(nums: IntArray): Long {
        var ans: Long = 0
        var cnt: Long = 0
        for (n in nums) {
            if (n == 0) cnt += 1
            else cnt = 0
            ans += cnt
        }
        return ans
    }
}

File: 2348. Number of Zero-Filled Subarrays.py
from typing import List
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        ans = cnt = 0
        nums.append(1)
        for n in nums:
            if n: cnt = 0
            else: cnt += 1
            ans += cnt
        return ans
